### PR TITLE
fix(docz): dependency issues

### PR DESCRIPTION
**Pull request by adbayb**
### Description

A PR to address several high priority issues. It been tested by emulating new npm release through a local registry (please note that if you found several package version updates in some commits: it's totally normal and I've cleaned all tested version metadata to prepare a future release through this commit https://github.com/adbayb/docz/commit/24027115300b9b79d1811558044fb71357bd9fc5).

However, @renatobenks It could be great if you can publish a new alpha version to let issue authors confirms their issue resolution 😄 

#### Main updates (fixes):
- [x] Fixes https://github.com/doczjs/docz/issues/1642 via https://github.com/adbayb/docz/commit/5f5a5d08efbad69616a38446202e06baa12812b2 and https://github.com/adbayb/docz/commit/abbf8beca9d34a40f21fdfdf928e969429700ba3 
- [x] Fixes https://github.com/doczjs/docz/issues/1635 via https://github.com/adbayb/docz/commit/5f5a5d08efbad69616a38446202e06baa12812b2  
I've fixed the `get-pkg-repo` to prevent incrementing to the broken patch version (hotfix docz side until a clean fix is done `get-pkg-repo` side)
- [x] Fixes https://github.com/pedronauck/gatsby-starter-docz/issues/4 via https://github.com/adbayb/docz/commit/e48cd4a01c53f5a3b47bd2549809a954e6baaf84
- [x] Fix unknown argument while running `gatsby` by preventing empty argument via https://github.com/adbayb/docz/commit/40dead8e84b643a21482a2531143aeb69431d694

👉 Some other issue reporting might be solved as well (I have not tested other issue resolution).

#### Other updates (chore):
- [x] Make dev environment works using monorepo capabilities via https://github.com/adbayb/docz/commit/e48cd4a01c53f5a3b47bd2549809a954e6baaf84 and https://github.com/adbayb/docz/commit/822a0b7bf138fa8a6b103a6304b1c2bf04e39bda
- [x] Clean some unused dependencies, collocate dependencies where they're needed and hoist dev dependencies to make local environment easier to maintain via https://github.com/adbayb/docz/commit/9ccdc44625c4e13e229936e5f2940a82c0f2a84e and https://github.com/adbayb/docz/commit/b33752748fa6c8bb28f944d773137be458c85651 
- [x] Fix lint errors via https://github.com/adbayb/docz/commit/a8fa53410ebcbe185059a21a763c2ba81b357b26

#### Opportunities (for future PRs)

- [ ] Update all dependencies (especially gatsby ones)
- [ ] Test with React 17 and update `peerDependencies` to accept newer version range

### CODE REVIEW

**Comment by renatobenks** [comment]
Awesome @adbayb, I'll gonna spend some time reviewing it, because I wanna stress some tests here, but great work!

So far I've seen, a couple issues are going to fix indeed!

**Comment by adbayb** [comment]
Hello @renatobenks 👋 , any update on this? 🙏

**Comment by renatobenks** [comment]
I'll be finishing some tests today, but once it's done, let's merge it, is that sounds good?

**Comment by adbayb** [comment]
Any news @renatobenks ?

**Comment by renatobenks** [comment]
Sorry for the big delay here, but I'm managing a new release for this one. I just need to ensure a few things first with @pedronauck, once I'm still not so sure about removing a few dependencies, even though it seems a good idea. But in general, so far so good in all my tests (which we need to automate ASAP from now on), and also, it'll be a great contribution for us all!

Thank you a lot @adbayb for the heads up on here 🙂

File: core/docz/package.json [hunk 1]
```diff
@@ -23,39 +23,20 @@
...
-    "gatsby-theme-docz": "^2.3.1",
+    "gatsby-theme-docz": "^2.3.1-alpha.0",
     "lodash": "^4.17.14",
     "marksy": "^8.0.0",
     "match-sorter": "^3.1.1",
     "pascal-case": "^3.1.0",
-    "prop-types": "^15.7.2",
-    "scheduler": "^0.15.0",
-    "ulid": "^2.3.0",
-    "yargs": "^13.3.0"
```
Comment (renatobenks) [inline core/docz/package.json:43]: keep this, please

**Review by renatobenks — CHANGES_REQUESTED** [review]
Hey, here follows the placed changes to rollback

**Comment by adbayb** [comment]
@renatobenks Done https://github.com/doczjs/docz/pull/1647/commits/90ab65d4376285fa9c7788fea6bcf51ffd7bbd86
Hope it'll help make your dependency review easier afterwards.

**Review by renatobenks — APPROVED** [review]

**Comment by adbayb** [comment]
@renatobenks Thank you for your review. Do you plan a release soon?

**Comment by renatobenks** [comment]
Yeap, I'm releasing today a new alpha version to the next patch, `v2.3.2` respectively, and in the meantime, you can be using this one until having the final patch version, which we're pretending to launch as soon as we finish a couple of automated tests that we're working on.

Let us know if you have any issues towards this end.

**Comment by adbayb** [comment]
Thank you @renatobenks 🙏 . 
Unfortunately, I can't find the newly alpha. It seems that the latest one is the `2.0.0-rc.23` (publication date: 02/09/2019).
What's the right version please? Thanks in advance.

Capture d’écran 2021-09-06 à 16 48 31
Capture d’écran 2021-09-06 à 16 46 57

**Comment by renatobenks** [comment]
Hey @adbayb, this upcoming version it's on https://github.com/doczjs/docz/tree/v2.3.3-alpha.0, still as alpha because I would like to get some people's feedback about how it goes, so let me know.

**Comment by adbayb** [comment]
Thanks @renatobenks , I've tested the alpha release: it works just fine in our company codebase 🚀

**Comment by drewlustro** [comment]
This is a lovely patch @adbayb, thank you for contributing it!

The `react-docgen` upgrade, in particular, made Prop table generation significantly more reliable for us.

Thanks again ❤️

**Pull request merged**